windows/src/desktop/kmshell/main/Keyman.System.UpdateStateMachine.pas

```diff
@@ -1017,19 +1017,20 @@ function InstallingState.DoInstallKeyman: Boolean;
 var
   FResult: Boolean;
   SavePath: String;
-  fileExt: String;
   FileName: String;
   FileNames: TStringDynArray;
+  ucr: TUpdateCheckResponse;
+  ucrFileName: String;
   found: Boolean;
 begin
-
+  TUpdateCheckStorage.LoadUpdateCacheData(ucr);
+  ucrFileName := ucr.FileName;
   SavePath := IncludeTrailingPathDelimiter(TKeymanPaths.KeymanUpdateCachePath);
   GetFileNamesInDirectory(SavePath, FileNames);
   found := False;
   for FileName in FileNames do
   begin
-    fileExt := LowerCase(ExtractFileExt(FileName));
-    if fileExt = '.exe' then
+    if ucrFileName = ExtractFileName(FileName) then
```

Comment:
```suggestion
    if SameText(ucrFileName, ExtractFileName(FileName)) then
```

Let's do the comparison case-insensitive to avoid possible case mismatches